components/brave_wallet_ui/page/screens/fund-wallet/deposit-funds.tsx

```diff
@@ -446,11 +448,17 @@ function DepositAccount() {
     selectedAsset ?? skipToken
   )
   const accountsForSelectedAssetCoinType = React.useMemo(() => {
-    return selectedAsset
-      ? accounts.filter((a) => a.accountId.coin === selectedAsset.coin)
-      : []
+    if (!selectedAssetNetwork) {
+      return []
+    }
+    return accounts.filter((a) =>
+      networkSupportsAccount(selectedAssetNetwork, a.accountId)
+    )
```

Comment:
pref nit: we can avoid flipping the boolean

```suggestion
    return selectedAssetNetwork ? accounts.filter((a) =>
      networkSupportsAccount(selectedAssetNetwork, a.accountId)
    ) : []
```

Reply:
https://github.com/brave/brave-core/pull/20812/commits/419eda221b83c47e8041e3eb0ef9d94a46c772e5